src/pages/test-report/cat-adi-part2/components/manoeuvres-popover/manoeuvres-popover.ts

```diff
@@ -40,34 +41,48 @@ export class ManoeuvresPopoverComponent {
       select(getTests),
       select(getCurrentTest),
       select(getTestData),
-      select(getManoeuvres),
+      select(getManoeuvresADI2),
     );
-    this.manoeuvresWithFaults$ = this.manoeuvres$.pipe(
-      // TO-DO ADI Part2: implement correct category
 
-      map((manoeuvres: CatBUniqueTypes.Manoeuvres) => ({
-        reverseRight: this.manoeuvreHasFaults(manoeuvres.reverseRight),
-        reverseParkRoad: this.manoeuvreHasFaults(manoeuvres.reverseParkRoad),
-        reverseParkCarpark: this.manoeuvreHasFaults(manoeuvres.reverseParkCarpark),
-        forwardPark: this.manoeuvreHasFaults(manoeuvres.forwardPark),
-      })),
+    this.manoeuvresWithFaults$ = this.manoeuvres$.pipe(
+      map((manoeuvres: CatADI2UniqueTypes.Manoeuvres[]) => {
+        return manoeuvres.map(manoeuvre => ({
+          reverseRight: this.manoeuvreHasFaults(manoeuvre.reverseRight),
+          reverseParkRoad: this.manoeuvreHasFaults(manoeuvre.reverseParkRoad),
+          reverseParkCarpark: this.manoeuvreHasFaults(manoeuvre.reverseParkCarpark),
+          forwardPark: this.manoeuvreHasFaults(manoeuvre.forwardPark),
+        }));
+      }),
     );
   }
 
-  recordManoeuvreSelection(manoeuvre: ManoeuvreTypes): void {
-    this.store$.dispatch(new RecordManoeuvresSelection(manoeuvre));
+  recordManoeuvreSelection(manoeuvre: ManoeuvreTypes, index: number): void {
+    this.store$.dispatch(new RecordManoeuvresSelection(manoeuvre, index));
   }
 
-  // TODO(MES-5031): implement as part of manoeuvres build
-  shouldManoeuvreDisable(manoeuvre: ManoeuvreTypes): Observable<boolean> {
-    return of(false);
-    // return this.manoeuvresWithFaults$.pipe(
-    //   map((manoeuvresWithFaults: ManoeuvresFaultState) => {
-    //     // TO-DO ADI Part2: implement correct category
-    //     const { [manoeuvre]: manoeuvreToOmit, ...otherManoeuvres } = manoeuvresWithFaults;
-    //     return some(otherManoeuvres, (value: boolean) => value);
-    //   }),
-    // );
+  shouldManoeuvreDisable(manoeuvre: ManoeuvreTypes, index: number): Observable<boolean> {
```

Comment:
Can't see any unit tests for `shouldManoeuvreDisable()` and `shouldHideManoeuvre()`, is the functionality they provide covered elsewhere?

Reply:
Same level of coverage as CatB!

Reply:
Don't think `shouldHideManoeuvre()` existed before ADI2

Reply:
You're right. I was joking because there's no coverage for any popover disabled logic in CatB either!

Reply:
Realised that only after I'd commented as I've just had a look! Not good at all 👎hopefully an anomaly and not a trend

Reply:
I'll add some tests for ADI2.

Reply:
Done and pushed.